fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PushPredicateThroughAggregation.java

```diff
@@ -0,0 +1,110 @@
+import org.apache.doris.nereids.rules.rewrite.OneRewriteRuleFactory;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.Slot;
+import org.apache.doris.nereids.trees.expressions.visitor.SlotExtractor;
+import org.apache.doris.nereids.trees.plans.GroupPlan;
+import org.apache.doris.nereids.trees.plans.Plan;
+import org.apache.doris.nereids.trees.plans.logical.LogicalAggregate;
+import org.apache.doris.nereids.trees.plans.logical.LogicalFilter;
+import org.apache.doris.nereids.util.ExpressionUtils;
+
+import com.google.common.collect.Lists;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+
+/**
+ * Push the predicate in the LogicalFilter to the aggregate child.
+ * For example:
+ * Logical plan tree:
+ *                 any_node
+ *                   |
+ *                filter (a>0 and b>0)
+ *                   |
+ *                group by(a, c)
+ *                   |
+ *                 scan
+ * transformed to:
+ *                 project
+ *                   |
+ *              upper filter (b>0)
+ *                   |
+ *                group by(a, c)
+ *                   |
+ *              bottom filter (a>0)
+ *                   |
+ *                 scan
+ * Note:
+ *    'a>0' could be push down, because 'a' is in group by keys;
+ *    but 'b>0' could not push down, because 'b' is not in group by keys.
+ *
+ */
+
+public class PushPredicateThroughAggregation extends OneRewriteRuleFactory {
+
+    @Override
+    public Rule build() {
+        return logicalFilter(logicalAggregate()).then(filter -> {
+            LogicalAggregate<GroupPlan> aggregate = filter.child();
+            Set<Slot> groupBySlots = new HashSet<>();
+            for (Expression groupByExpression : aggregate.getGroupByExpressionList()) {
+                if (groupByExpression instanceof Slot) {
+                    groupBySlots.add((Slot) groupByExpression);
+                }
+            }
+            List<Expression> pushDownPredicates = Lists.newArrayList();
+            List<Expression> filterPredicates = Lists.newArrayList();
+            ExpressionUtils.extractConjunct(filter.getPredicates()).forEach(conjunct -> {
+                Set<Slot> conjunctSlots = SlotExtractor.extractSlot(conjunct);
+                if (groupBySlots.containsAll(conjunctSlots)) {
+                    pushDownPredicates.add(conjunct);
+                } else {
+                    //TODO here
+                    filterPredicates.add(conjunct);
+                }
+            });
+
+            return pushDownPredicate(filter, aggregate, pushDownPredicates, filterPredicates);
+        }).toRule(RuleType.PUSH_DOWN_PREDICATE_THROUGH_AGGREGATION);
+    }
+
+    private Plan pushDownPredicate(LogicalFilter filter, LogicalAggregate aggregate,
+                                   List<Expression> pushDownPredicates, List<Expression> filterPredicates) {
+        Plan root;
```

Comment:
nit: directly return the local result in the `if  else` branch?

Reply:
If we keep only one return in a function, it would be more readable.
Anyway, this is more like a code format style. If our team has such rule, I will follow.

Reply:
Yeah, it's about code style. Any thoughts about this? @924060929 @morrySnow @EmmyMiao87?

Reply:
@924060929 here we discussed your concern.

Reply:
code was refactored, add `return` in if-else clause